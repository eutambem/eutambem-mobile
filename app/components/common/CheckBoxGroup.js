import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';

import { DARK_PURPLE } from '../../styles';
import CheckBox from './CheckBox';

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    marginBottom: 10,
  },
  text: {
    fontSize: 14,
    color: DARK_PURPLE,
  },
  checkbox: {
    paddingTop: 10,
  },
});

export default class CheckBoxGroup extends Component {
  constructor(props) {
    super(props);
    const values = {};
    props.options.forEach((option) => { values[option.value] = false; });
    this.state = { values };
  }

  onClick(key) {
    const { values } = this.state;
    values[key] = !values[key];
    this.setState({ values });
  }

  renderOptions() {
    return this.props.options.map(option => (
      <CheckBox
        key={option.value}
        onClick={() => this.onClick(option.value)}
        isChecked={this.state.values[option.value]}
        label={option.label}
      />
    ));
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{this.props.label}</Text>
        {this.renderOptions()}
      </View>
    );
  }
}

CheckBoxGroup.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    key: PropTypes.string,
  })).isRequired,
  label: PropTypes.string.isRequired,
};
