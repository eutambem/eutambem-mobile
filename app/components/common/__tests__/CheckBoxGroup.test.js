import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import CheckBoxGroup from '../CheckBoxGroup';

const props = {
  label: 'Test CheckBoxGroup',
  options: [{
    key: '1',
    label: 'Label 1',
  }, {
    key: '2',
    label: 'Label 2',
  }],
};

describe('CheckBoxGroup', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CheckBoxGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
