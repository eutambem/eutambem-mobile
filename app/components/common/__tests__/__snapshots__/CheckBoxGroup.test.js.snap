// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ForwardRef(Text)
    style={
      Object {
        "color": "#961687",
        "fontSize": 14,
      }
    }
  >
    Test CheckBoxGroup
  </ForwardRef(Text)>
  <CheckBox
    key="1"
    label="Label 1"
    onClick={[Function]}
  />
  <CheckBox
    key="2"
    label="Label 2"
    onClick={[Function]}
  />
</ForwardRef(View)>
`;
